src/io/iohk/scalanet/peergroup/SimplePeerGroup.scala

```diff
@@ -59,43 +60,41 @@ class SimplePeerGroup[A, F[_], AA](
   // TODO if no known peers, create a default routing table with just me.
   // TODO otherwise, enroll with one or more known peers (and obtain/install their routing table here).
 
-  override def sendMessage(address: A, message: ByteBuffer): F[Unit] = {
+  override def sendMessage(address: A, message: ByteBuffer): Task[Unit] = {
     // TODO if necessary frame the buffer with peer group specific fields
     // Lookup A in the routing table to obtain an AA for the underlying group.
     // Call sendMessage on the underlyingPeerGroup
     val underLyingAddress = routingTable(address)
     underLyingPeerGroup.sendMessage(underLyingAddress, message)
   }
 
-  override def shutdown(): F[Unit] = underLyingPeerGroup.shutdown()
+  override def shutdown(): Task[Unit] = underLyingPeerGroup.shutdown()
 
-  override def initialize(): F[Unit] = {
+  override def initialize(): Task[Unit] = {
     routingTable += processAddress -> underLyingPeerGroup.processAddress
 
     if (config.knownPeers.nonEmpty) {
       val (knownPeerAddress, knownPeerAddressUnderlying) = config.knownPeers.head
       routingTable += knownPeerAddress -> knownPeerAddressUnderlying
 
-      controlChannel.sendMessage(
-        knownPeerAddressUnderlying,
-        EnrolMe(config.processAddress, underLyingPeerGroup.processAddress)
-      )
-
-      val enrolledTask: Task[Unit] = Task.deferFutureAction { implicit scheduler =>
-        controlChannel.inboundMessages
-          .collect {
-            case Enrolled(_, _, newRoutingTable) =>
-              routingTable.clear()
-              routingTable ++= newRoutingTable
-              log.debug(s"Peer address '$processAddress' enrolled into group and installed new routing table:")
-              log.debug(s"$newRoutingTable")
-          }
-          .head()
-      }
-
-      liftF(enrolledTask)
+      val enrolledTask: Task[Unit] = controlChannel.inboundMessages.collect {
+        case Enrolled(_, _, newRoutingTable) =>
+          routingTable.clear()
+          routingTable ++= newRoutingTable
+          log.debug(s"Peer address '$processAddress' enrolled into group and installed new routing table:")
+          log.debug(s"$newRoutingTable")
```

Comment:
Please, merge the logs to find these details easily on concurrent environments.

Reply:
Done. I guess in fact correlation ids are the only way to make sense of logs in concurrent environments.